File: src/statesman/commands/PrintRandomCommand.java
package statesman.commands;

import java.util.Random;

public class PrintRandomCommand extends PrintCombineCommand {

    public static final String ID = "printr";

    private Random _random;

    public PrintRandomCommand() {
        super();
        _random = new Random();
    }

    public PrintRandomCommand(String[] messages) {
        this();
        _messages = messages;
    }

    @Override
    public Command createInstance(String[] arguments) {
        if (arguments.length > 2) {
            String[] messages = new String[arguments.length - 1];
            System.arraycopy(arguments, 1, messages, 0, arguments.length - 1);
            return new PrintRandomCommand(messages);
        }
        return null;
    }

    @Override
    public void execute() {
        initializeStrings();
        int i = _random.nextInt(_messages.length);
        System.out.printf(_messages[i] + "%n");
    }

}

File: src/Commands/PrintRandomCommand.cs
using System;

namespace Statesman.Commands
{
    public class PrintRandomCommand : PrintCombineCommand
    {
        public static readonly string ID = "printr";

        private readonly Random _random;

        public PrintRandomCommand()
            : base()
        {
            _random = new Random();
        }

        public PrintRandomCommand(string[] messages)
            : this()
        {
            _messages = messages;
        }

        public override Command CreateInstance(string[] arguments)
        {
            if (arguments.Length > 2)
            {
                string[] messages = new string[arguments.Length - 1];
                Array.Copy(arguments, 1, messages, 0, arguments.Length - 1);
                return new PrintRandomCommand(messages);
            }
            return null;
        }

        public override void Execute()
        {
            InitializeStrings();
            int i = _random.Next(_messages.Length);
            Console.WriteLine(_messages[i]);
        }
    }
}
